### PR TITLE
UIPO-88 Feature/register form component

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/components/LoginPage/LoginPage.js [hunk 1]
```diff
@@ -134,12 +135,14 @@ class LoginPage extends React.Component {
...
-            create an account
-            </Button>
+            <Link to="/RegisterForm">
+              <Button
+                className="black-button"
+                onClick={this.validate}
+              >
+              create an account
+              </Button>
+            </Link>
```
Comment (Hollister009) [inline src/components/LoginPage/LoginPage.js:145]: well this incorrect `<Link>` is already the `<a>` tag under the hood

File: src/components/RegisterForm/RegisterForm.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    this.setState({
+      email: ls.get('emailToRegister') || ''
+    });
+  }
+
+  saveToLocalStore() {
+
+  }
+
+  handleSubmit(e) {
```
Comment (mirafedas) [inline src/components/RegisterForm/RegisterForm.js:33]: You can use destructuring here, like this: 
```jsx
handleSubmit({ target }) {
    e.preventDefault();
    const form = target;
    const data = new FormData(form);
(...)
}
```

File: src/components/RegisterForm/RegisterForm.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    e.preventDefault();
+    const form = e.target;
+    const data = new FormData(form);
+
+    fetch('/api/form-submit-url', {
```
Comment (mirafedas) [inline src/components/RegisterForm/RegisterForm.js]: Please move '/api/form-submit-url' to the constant (like this: const URL = '/api/form-submit-url'; fetch(URL, {...})

File: src/components/LoginPage/LoginPage.js (hunk 1 above)
Comment (Hollister009) [reply src/components/LoginPage/LoginPage.js:145]: better to write something like this:
```jsx
const RegisterLink = 
  <Button className="black-button" onClick={this.validate}>
    create an account
  </Button>;

<Link to="/register-form" component={RegisterLink} />
```

**Comment by Hollister009** [comment]
@myrosvas take a look!

File: src/components/RegisterForm/RegisterForm.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      method: 'POST',
+      body: data
+    });
+  }
+
+  render() {
+    const { email } = this.state;
+
+    return (
+      <div className={`${CN}`} onSubmit={this.handleSubmit}>
```
Comment (mirafedas) [inline src/components/RegisterForm/RegisterForm.js]: It is a bad practice to set onSubmit on non-interactive elements (like div). I suggest to change the HTML structure to this one:
form onSubmit={this.handleSubmit}
   div Personal info /div
   div Shipping info /div
/form

File: src/components/RegisterForm/RegisterForm.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+            </div>
+
+            <div className={`${CN}__field`}>
+              <label className={`${CN}__field__label`}>last name*</label>
+              <input className={`${CN}__field__input`} type="text" required />
+            </div>
+
+            <div className={`${CN}__field`}>
+              <label className={`${CN}__field__label`}>e-mail*</label>
+              <input className={`${CN}__field__input`} type="email" id="email" value={email} required />
```
Comment (mirafedas) [inline src/components/RegisterForm/RegisterForm.js]: When there is more than 3 properties, it is better to start each property in a new row to make it more readable, like this:
<input
 className={`${CN}__field__input`}
 type="email"
 id="email"
 value={email}
 required
/>

**Review by mirafedas — CHANGES_REQUESTED** [review]

**Review by mirafedas — APPROVED** [review]

File: src/components/RegisterForm/RegisterForm.js (hunk 2 above)
Comment (Hollister009) [reply src/components/RegisterForm/RegisterForm.js]: @myrosvas in your case you lost the reference to e variable

**Review by Hollister009 — APPROVED** [review]
Approved

File: src/components/RegisterForm/RegisterForm.js (hunk 2 above)
Comment (Hollister009) [reply src/components/RegisterForm/RegisterForm.js:33]: not true, in this case, you are loosening reference to **e** variable

**Pull request merged**